presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -100,11 +101,21 @@ public void getAllQueryInfo(
             proxyResponse(servletRequest, asyncResponse, xForwardedProto, uriInfo);
             return;
         }
+        if (limitFilter == null) {
```

Comment:
Do we want to check the argument to be positive when it is not null?

Reply:
Mmm, we can, of course and what do we do if it is not positive? Do we accept zeros?

Reply:
I think we should throw `IllegalArgumentException` and return either a 400 or 422 to the user?